adapters/repos/db/replication.go

```diff
@@ -476,8 +476,8 @@ func (db *DB) FetchObjects(ctx context.Context,
 func (i *Index) fetchObjects(ctx context.Context,
 	shardName string, ids []strfmt.UUID,
 ) ([]objects.Replica, error) {
-	shard := retryGetLocalShard(i, shardName)
-	if shard == nil {
+	shard, err := i.getOrInitLocalShard(ctx, shardName)
```

Comment:
I believe we need to recreate the shard only in the case of a write request. Read requests should return an empty result if a shard does not exist.